contracts/MechMarketplace.sol

```diff
@@ -122,44 +164,51 @@ contract MechMarketplace {
     bytes32 public immutable domainSeparator;
     // Original chain Id
     uint256 public immutable chainId;
+    // Minimum response time
+    uint256 public immutable minResponseTimeout;
+    // Maximum response time
+    uint256 public immutable maxResponseTimeout;
+    // Approved mech bytecode hash
+    bytes32 public immutable mechBytecodeHash;
```

Comment:
Specifying mech contract bytecode hash this marketplace is going to work with in order to avoid malicious mechs.

Reply:
Hm, I see, this answers my question below. But this also means the marketplace is static wrt to mechs.

What exact exploit are we preventing with this?

Reply:
We have elaborated on this already, changing in the next PR to deal with the staking factory verification for both mech and trader staking contracts.